### PR TITLE
runtime: incorrect handling of non-empty []Endpoint parameter in Remo…

### PR DESCRIPTION
**Pull request by Xuanqing-Shi**
…veEndpoints function

In the RemoveEndpoints(), when the endpoints paramete isn't empty, using idx may result in wrong endpoint removals. To improve, directly passing the endpoint parameter helps locate the correct elements within n.eps.

Fixes: #7732

### CODE REVIEW

File: src/runtime/virtcontainers/network_linux.go [hunk 1]
```diff
@@ -211,13 +211,18 @@ func (n *LinuxNetwork) addSingleEndpoint(ctx context.Context, s *Sandbox, netInf
...
+	var idx int = len(n.eps)
+	for i, val := range n.eps {
+		if val == endpoint {
+			idx = i
+			break
+		}
+	}
 	if idx > len(n.eps)-1 {
 		return fmt.Errorf("Endpoint index overflow")
 	}
```
Comment (justxuewei) [inline src/runtime/virtcontainers/network_linux.go:224]: To make it clear, I tend to not to use `idx > len(n.eps)-1`.

```suggestion
	if idx == len(n.eps) {
		return fmt.Errorf("Endpoint not found")
	}
```

File: src/runtime/virtcontainers/network_linux.go [hunk 2]
```diff
@@ -211,13 +211,18 @@ func (n *LinuxNetwork) addSingleEndpoint(ctx context.Context, s *Sandbox, netInf
 	return endpoint, nil
 }
 
-func (n *LinuxNetwork) removeSingleEndpoint(ctx context.Context, s *Sandbox, idx int, hotplug bool) error {
+func (n *LinuxNetwork) removeSingleEndpoint(ctx context.Context, s *Sandbox, endpoint Endpoint, hotplug bool) error {
+	var idx int = len(n.eps)
+	for i, val := range n.eps {
+		if val == endpoint {
```
Comment (bergwolf) [inline src/runtime/virtcontainers/network_linux.go:217]: Looking at [`findEndpoint`](https://github.com/kata-containers/kata-containers/blob/main/src/runtime/virtcontainers/endpoint.go#L232), we are relying on the endpoint hardware address to determine if two endpoints are the same. So here we should use
```
if val.HardwareAddr() == endpoint.HardwareAddr() {
```

File: src/runtime/virtcontainers/network_linux.go (hunk 1 above)
Comment (Xuanqing-Shi) [reply src/runtime/virtcontainers/network_linux.go]: Thanks for your feedback! I agree with your suggestion to modify the original source code. I'll make the necessary changes in the PR.

File: src/runtime/virtcontainers/network_linux.go (hunk 2 above)
Comment (Xuanqing-Shi) [reply src/runtime/virtcontainers/network_linux.go]: Thank you for your valuable feedback and suggestions. I have made the necessary modifications to address your concerns. You can review the changes in the updated.

**Review by bergwolf — APPROVED** [review]
lgtm, thanks!

**Comment by bergwolf** [comment]
@Xuanqing-Shi Please squash your commits into a single one, and add commit message as explained in https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#patch-format

**Review by teawater — APPROVED** [review]

**Pull request merged**